## YASHASVI VERMA

### Exploratory data analysis project

Data Cleaning and EDA using the IMDB movies dataset  

Importing python libraries

In [2]:
import pandas as pd

In [3]:
import numpy as np

In [4]:
df = pd.read_csv('C:\Datasets\movies.csv')

# Cleaning the data

In [11]:
df

budget                                             genres  \
0     237000000  [{"id": 28, "name": "Action"}, {"id": 12, "nam...   
1     300000000  [{"id": 12, "name": "Adventure"}, {"id": 14, "...   
2     245000000  [{"id": 28, "name": "Action"}, {"id": 12, "nam...   
3     250000000  [{"id": 28, "name": "Action"}, {"id": 80, "nam...   
4     260000000  [{"id": 28, "name": "Action"}, {"id": 12, "nam...   
...         ...                                                ...   
4798     220000  [{"id": 28, "name": "Action"}, {"id": 80, "nam...   
4799       9000  [{"id": 35, "name": "Comedy"}, {"id": 10749, "...   
4800          0  [{"id": 35, "name": "Comedy"}, {"id": 18, "nam...   
4801          0                                                 []   
4802          0                [{"id": 99, "name": "Documentary"}]   

                                               homepage      id  \
0                           http://www.avatarmovie.com/   19995   
1          http://disney.go.com/disneypictures/pirates/     285   
2           http://www.sonypictures.com/movies/spectre/  206647   
3                    http://www.thedarkknightrises.com/   49026   
4                  http://movies.disney.com/john-carter   49529   
...                                                 ...     ...   
4798                                                NaN    9367   
4799                                                NaN   72766   
4800  http://www.hallmarkchannel.com/signedsealeddel...  231617   
4801                        http://shanghaicalling.com/  126186   
4802                                                NaN   25975   

                                               keywords original_language  \
0     [{"id": 1463, "name": "culture clash"}, {"id":...                en   
1     [{"id": 270, "name": "ocean"}, {"id": 726, "na...                en   
2     [{"id": 470, "name": "spy"}, {"id": 818, "name...                en   
3     [{"id": 849, "name": "dc comics"}, {"id": 853,...                en   
4     [{"id": 818, "name": "based on novel"}, {"id":...                en   
...                                                 ...               ...   
4798  [{"id": 5616, "name": "united states\u2013mexi...                es   
4799                                                 []                en   
4800  [{"id": 248, "name": "date"}, {"id": 699, "nam...                en   
4801                                                 []                en   
4802  [{"id": 1523, "name": "obsession"}, {"id": 224...                en   

                                original_title  \
0                                       Avatar   
1     Pirates of the Caribbean: At World's End   
2                                      Spectre   
3                        The Dark Knight Rises   
4                                  John Carter   
...                                        ...   
4798                               El Mariachi   
4799                                 Newlyweds   
4800                 Signed, Sealed, Delivered   
4801                          Shanghai Calling   
4802                         My Date with Drew   

                                               overview  popularity  \
0     In the 22nd century, a paraplegic Marine is di...  150.437577   
1     Captain Barbossa, long believed to be dead, ha...  139.082615   
2     A cryptic message from Bond’s past sends him o...  107.376788   
3     Following the death of District Attorney Harve...  112.312950   
4     John Carter is a war-weary, former military ca...   43.926995   
...                                                 ...         ...   
4798  El Mariachi just wants to play his guitar and ...   14.269792   
4799  A newlywed couple's honeymoon is upended by th...    0.642552   
4800  "Signed, Sealed, Delivered" introduces a dedic...    1.444476   
4801  When ambitious New York attorney Sam is sent t...    0.857008   
4802  Ever since the second grade when he first saw ... 

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4803 entries, 0 to 4802
Data columns (total 20 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   budget                4803 non-null   int64  
 1   genres                4803 non-null   object 
 2   homepage              1712 non-null   object 
 3   id                    4803 non-null   int64  
 4   keywords              4803 non-null   object 
 5   original_language     4803 non-null   object 
 6   original_title        4803 non-null   object 
 7   overview              4800 non-null   object 
 8   popularity            4803 non-null   float64
 9   production_companies  4803 non-null   object 
 10  production_countries  4803 non-null   object 
 11  release_date          4802 non-null   object 
 12  revenue               4803 non-null   int64  
 13  runtime               4801 non-null   float64
 14  spoken_languages      4803 non-null   object 
 15  status               

In [9]:
df.describe()

budget             id   popularity       revenue      runtime  \
count  4.803000e+03    4803.000000  4803.000000  4.803000e+03  4801.000000   
mean   2.904504e+07   57165.484281    21.492301  8.226064e+07   106.875859   
std    4.072239e+07   88694.614033    31.816650  1.628571e+08    22.611935   
min    0.000000e+00       5.000000     0.000000  0.000000e+00     0.000000   
25%    7.900000e+05    9014.500000     4.668070  0.000000e+00    94.000000   
50%    1.500000e+07   14629.000000    12.921594  1.917000e+07   103.000000   
75%    4.000000e+07   58610.500000    28.313505  9.291719e+07   118.000000   
max    3.800000e+08  459488.000000   875.581305  2.787965e+09   338.000000   

       vote_average    vote_count  
count   4803.000000   4803.000000  
mean       6.092172    690.217989  
std        1.194612   1234.585891  
min        0.000000      0.000000  
25%        5.600000     54.000000  
50%        6.200000    235.000000  
75%        6.800000    737.000000  
max       10.000000  13752.000000

In [10]:
df.isnull().sum().sort_values()

budget                     0
title                      0
status                     0
spoken_languages           0
revenue                    0
production_countries       0
vote_average               0
popularity                 0
production_companies       0
original_title             0
original_language          0
keywords                   0
id                         0
genres                     0
vote_count                 0
release_date               1
runtime                    2
overview                   3
tagline                  844
homepage                3091
dtype: int64

In [90]:
df = df.drop(['homepage','tagline','overview'], axis=1)

In [91]:
df.head(2)

budget                                             genres     id  \
0  237000000  [{"id": 28, "name": "Action"}, {"id": 12, "nam...  19995   
1  300000000  [{"id": 12, "name": "Adventure"}, {"id": 14, "...    285   

                                            keywords original_language  \
0  [{"id": 1463, "name": "culture clash"}, {"id":...                en   
1  [{"id": 270, "name": "ocean"}, {"id": 726, "na...                en   

                             original_title  popularity  \
0                                    Avatar  150.437577   
1  Pirates of the Caribbean: At World's End  139.082615   

                                production_companies  \
0  [{"name": "Ingenious Film Partners", "id": 289...   
1  [{"name": "Walt Disney Pictures", "id": 2}, {"...   

                                production_countries release_date     revenue  \
0  [{"iso_3166_1": "US", "name": "United States o...   2009-12-10  2787965087   
1  [{"iso_3166_1": "US", "name": "United States o...   2007-05-19   961000000   

   runtime                                   spoken_languages    status  \
0    162.0  [{"iso_639_1": "en", "name": "English"}, {"iso...  Released   
1    169.0           [{"iso_639_1": "en", "name": "English"}]  Released   

                                      title  vote_average  vote_count  
0                                    Avatar           7.2       11800  
1  Pirates of the Caribbean: At World's End           6.9        4500

In [92]:
print(df[df["release_date"].isnull()])

      budget genres      id keywords original_language  \
4553       0     []  380097       []                en   

                  original_title  popularity production_companies  \
4553  America Is Still the Place         0.0                   []   

     production_countries release_date  revenue  runtime spoken_languages  \
4553                   []          NaN        0      0.0               []   

        status                       title  vote_average  vote_count  
4553  Released  America Is Still the Place           0.0           0  


In [93]:
df = df[df['release_date'].notnull()]

In [94]:
df.isnull().sum().sort_values(ascending=False)

runtime                 2
vote_count              0
production_companies    0
genres                  0
id                      0
keywords                0
original_language       0
original_title          0
popularity              0
production_countries    0
vote_average            0
release_date            0
revenue                 0
spoken_languages        0
status                  0
title                   0
budget                  0
dtype: int64

In [95]:
df.groupby('runtime').runtime.count()

runtime
0.0      34
14.0      1
25.0      1
41.0      1
42.0      1
         ..
242.0     1
248.0     1
254.0     1
276.0     1
338.0     1
Name: runtime, Length: 156, dtype: int64

In [96]:
df.runtime.describe()

count    4800.000000
mean      106.898125
std        22.561593
min         0.000000
25%        94.000000
50%       103.000000
75%       118.000000
max       338.000000
Name: runtime, dtype: float64

In [97]:
df.runtime = df.runtime.fillna(106.8)

In [98]:
df.isnull().sum().sort_values(ascending=False)

vote_count              0
production_companies    0
genres                  0
id                      0
keywords                0
original_language       0
original_title          0
popularity              0
production_countries    0
vote_average            0
release_date            0
revenue                 0
runtime                 0
spoken_languages        0
status                  0
title                   0
budget                  0
dtype: int64

In [99]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4802 entries, 0 to 4802
Data columns (total 17 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   budget                4802 non-null   int64  
 1   genres                4802 non-null   object 
 2   id                    4802 non-null   int64  
 3   keywords              4802 non-null   object 
 4   original_language     4802 non-null   object 
 5   original_title        4802 non-null   object 
 6   popularity            4802 non-null   float64
 7   production_companies  4802 non-null   object 
 8   production_countries  4802 non-null   object 
 9   release_date          4802 non-null   object 
 10  revenue               4802 non-null   int64  
 11  runtime               4802 non-null   float64
 12  spoken_languages      4802 non-null   object 
 13  status                4802 non-null   object 
 14  title                 4802 non-null   object 
 15  vote_average         

# 1) Which are the movies with the third-lowest and third-highest budget?

In [133]:
a = df.sort_values(by=['budget'])

In [136]:
a

budget                                             genres      id  \
4802          0                [{"id": 99, "name": "Documentary"}]   25975   
3582          0                      [{"id": 18, "name": "Drama"}]   15660   
3585          0                      [{"id": 18, "name": "Drama"}]   19848   
3587          0  [{"id": 18, "name": "Drama"}, {"id": 53, "name...  235271   
1261          0                     [{"id": 35, "name": "Comedy"}]   11025   
...         ...                                                ...     ...   
4     260000000  [{"id": 28, "name": "Action"}, {"id": 12, "nam...   49529   
10    270000000  [{"id": 12, "name": "Adventure"}, {"id": 14, "...    1452   
7     280000000  [{"id": 28, "name": "Action"}, {"id": 12, "nam...   99861   
1     300000000  [{"id": 12, "name": "Adventure"}, {"id": 14, "...     285   
17    380000000  [{"id": 12, "name": "Adventure"}, {"id": 28, "...    1865   

                                               keywords original_language  \
4802  [{"id": 1523, "name": "obsession"}, {"id": 224...                en   
3582  [{"id": 3598, "name": "ax"}, {"id": 5565, "nam...                en   
3585  [{"id": 3688, "name": "strip club"}, {"id": 65...                en   
3587  [{"id": 1326, "name": "infidelity"}, {"id": 15...                en   
1261  [{"id": 1608, "name": "twin sister"}, {"id": 5...                en   
...                                                 ...               ...   
4     [{"id": 818, "name": "based on novel"}, {"id":...                en   
10    [{"id": 83, "name": "saving the world"}, {"id"...                en   
7     [{"id": 8828, "name": "marvel comic"}, {"id": ...                en   
1     [{"id": 270, "name": "ocean"}, {"id": 726, "na...                en   
17    [{"id": 658, "name": "sea"}, {"id": 1316, "nam...                en   

                                   original_title  popularity  \
4802                            My Date with Drew    1.929883   
3582                               Mommie Dearest    3.102862   
3585                             The Players Club    5.769822   
3587                                     Addicted    7.544515   
1261                              New York Minute   10.040300   
...                                           ...         ...   
4                                     John Carter   43.926995   
10                               Superman Returns   57.925623   
7                         Avengers: Age of Ultron  134.279229   
1        Pirates of the Caribbean: At World's End  139.082615   
17    Pirates of the Caribbean: On Stranger Tides  135.413856   

                                   production_companies  \
4802  [{"name": "rusty bear entertainment", "id": 87...   
3582          [{"name": "Paramount Pictures", "id": 4}]   
3585            [{"name": "New Line Cinema", "id": 12}]   
3587  [{"name": "Lionsgate", "id": 1632}, {"name": "...   
1261  [{"name": "DiNovi Pictures", "id": 813}, {"nam...   
...                                                 ...   
4           [{"name": "Walt Disney Pictures", "id": 2}]   
10    [{"name": "DC Comics", "id": 429}, {"name": "L...   
7     [{"name": "Marvel Studios", "id": 420}, {"name...   
1     [{"name": "Walt Disney Pictures", "id": 2}, {"...   
17    [{"name": "Walt Disney Pictures", "id": 2}, {"...   

                                   production_countries release_date  \
4802  [{"iso_3166_1": "US", "name": "United States o...   2005-08-05   
3582  [{"iso_3166_1": "US", "name": "United States o...   1981-09-18   
3585  [{"iso_3166_1": "US", "name": "United States o...   1998-04-08   
3587  [{"iso_3166_1": "US", "name": "United States o...   2014-10-10   
1261  [{"iso_3166_1": "US", "name": "United States o...   2004-05-01   
...                                                 ...          ...   
4     [{"iso_3166_1": "US", "name": "United States o...   2012-03-07   
10    [{"iso_3166_1": "US", "name": "United States o...   2006-06-28   
7     [{"

In [102]:
a[2:3].original_title

3585    The Players Club
Name: original_title, dtype: object

In [103]:
a[-3:-2].original_title

7    Avengers: Age of Ultron
Name: original_title, dtype: object

ANS : Movie with the third-lowest budget is "The Players Club"
Movie with third-highest budget is "Avengers: Age of Ultron"

# 2) What is the average number of words in movie titles between the years 2000-2005?

In [104]:
df.head(2)

budget                                             genres     id  \
0  237000000  [{"id": 28, "name": "Action"}, {"id": 12, "nam...  19995   
1  300000000  [{"id": 12, "name": "Adventure"}, {"id": 14, "...    285   

                                            keywords original_language  \
0  [{"id": 1463, "name": "culture clash"}, {"id":...                en   
1  [{"id": 270, "name": "ocean"}, {"id": 726, "na...                en   

                             original_title  popularity  \
0                                    Avatar  150.437577   
1  Pirates of the Caribbean: At World's End  139.082615   

                                production_companies  \
0  [{"name": "Ingenious Film Partners", "id": 289...   
1  [{"name": "Walt Disney Pictures", "id": 2}, {"...   

                                production_countries release_date     revenue  \
0  [{"iso_3166_1": "US", "name": "United States o...   2009-12-10  2787965087   
1  [{"iso_3166_1": "US", "name": "United States o...   2007-05-19   961000000   

   runtime                                   spoken_languages    status  \
0    162.0  [{"iso_639_1": "en", "name": "English"}, {"iso...  Released   
1    169.0           [{"iso_639_1": "en", "name": "English"}]  Released   

                                      title  vote_average  vote_count  
0                                    Avatar           7.2       11800  
1  Pirates of the Caribbean: At World's End           6.9        4500

In [137]:
df.original_title

0                                         Avatar
1       Pirates of the Caribbean: At World's End
2                                        Spectre
3                          The Dark Knight Rises
4                                    John Carter
                          ...                   
4798                                 El Mariachi
4799                                   Newlyweds
4800                   Signed, Sealed, Delivered
4801                            Shanghai Calling
4802                           My Date with Drew
Name: original_title, Length: 4802, dtype: object

In [105]:
start_date = "2000-01-01"
end_date = "2005-12-31"

In [106]:
a = df['release_date'] >= start_date

In [107]:
b = df['release_date'] <= end_date

In [108]:
between_2000_5 = a & b

In [109]:
filter_data = df.loc[between_2000_5]

In [110]:
new_data = filter_data

In [111]:
new_data

budget                                             genres      id  \
24    207000000  [{"id": 12, "name": "Adventure"}, {"id": 18, "...     254   
30    200000000  [{"id": 28, "name": "Action"}, {"id": 12, "nam...     558   
63    180000000  [{"id": 12, "name": "Adventure"}, {"id": 10751...     411   
90    165000000  [{"id": 12, "name": "Adventure"}, {"id": 16, "...    5255   
93    200000000  [{"id": 28, "name": "Action"}, {"id": 53, "nam...     296   
...         ...                                                ...     ...   
4790          0  [{"id": 18, "name": "Drama"}, {"id": 10769, "n...   13898   
4793          0                      [{"id": 18, "name": "Drama"}]  182291   
4796       7000  [{"id": 878, "name": "Science Fiction"}, {"id"...   14337   
4797          0  [{"id": 10769, "name": "Foreign"}, {"id": 53, ...   67238   
4802          0                [{"id": 99, "name": "Documentary"}]   25975   

                                               keywords original_language  \
24    [{"id": 774, "name": "film business"}, {"id": ...                en   
30    [{"id": 851, "name": "dual identity"}, {"id": ...                en   
63    [{"id": 83, "name": "saving the world"}, {"id"...                en   
90    [{"id": 1991, "name": "santa claus"}, {"id": 5...                en   
93    [{"id": 83, "name": "saving the world"}, {"id"...                en   
...                                                 ...               ...   
4790                                                 []                fa   
4793  [{"id": 718, "name": "confession"}, {"id": 100...                en   
4796  [{"id": 1448, "name": "distrust"}, {"id": 2101...                en   
4797                                                 []                en   
4802  [{"id": 1523, "name": "obsession"}, {"id": 224...                en   

                                         original_title  popularity  \
24                                            King Kong   61.226010   
30                                         Spider-Man 2   35.149586   
63    The Chronicles of Narnia: The Lion, the Witch ...   67.391328   
90                                    The Polar Express   47.323228   
93                   Terminator 3: Rise of the Machines   69.405188   
...                                                 ...         ...   
4790                                              دایره    1.193779   
4793                                     On The Downlow    0.029757   
4796                                             Primer   23.307949   
4797                                             Cavite    0.022173   
4802                                  My Date with Drew    1.929883   

                                   production_companies  \
24    [{"name": "WingNut Films", "id": 11}, {"name":...   
30    [{"name": "Columbia Pictures", "id": 5}, {"nam...   
63    [{"name": "Walt Disney", "id": 5888}, {"name":...   
90    [{"name": "Golden Mean", "id": 1867}, {"name":...   
93    [{"name": "Columbia Pictures", "id": 5}, {"nam...   
...                                                 ...   
4790  [{"name": "Jafar Panahi Film Productions", "id...   
4793        [{"name": "Iconoclast Films", "id": 26677}]   
4796                 [{"name": "Thinkfilm", "id": 446}]   
4797                                                 []   
4802  [{"name": "rusty bear entertainment", "id": 87...   

                                   production_countries release_date  \
24    [{"iso_3166_1": "NZ", "name": "New Zealand"}, ...   2005-12-14   
30    [{"iso_3166_1": "US", "name": "United States o...   2004-06-25   
63    [{"iso_3166_1": "GB", "name": "United Kingdom"...   2005-12-07   
90    [{"iso_3166_1": "US", "name": "United States o...   2004-11-10   
93    [{"iso_3166_1": "DE", "name": "Germany"}, {"is...   2003-07-02   
...                                                 ...          ...   
4790             [{"iso_3166_1": "IR", "name": "Iran"}]   2000-09-08   
4793  [{"

In [138]:
new_data.original_title

24                                              King Kong
30                                           Spider-Man 2
63      The Chronicles of Narnia: The Lion, the Witch ...
90                                      The Polar Express
93                     Terminator 3: Rise of the Machines
                              ...                        
4790                                                دایره
4793                                       On The Downlow
4796                                               Primer
4797                                               Cavite
4802                                    My Date with Drew
Name: original_title, Length: 1142, dtype: object

In [112]:
count = new_data['original_title'].str.split().str.len()

In [113]:
count.sort_index(inplace=False)

24       2
30       2
63      11
90       3
93       6
        ..
4790     1
4793     3
4796     1
4797     1
4802     4
Name: original_title, Length: 1142, dtype: int64

In [114]:
count.mean()

2.7425569176882663

ANS : The average number of words in movie titles between the years 2000-2005 are *2.7425569176882663*

# 4) Which are the movies with the most and least earned revenue?

In [115]:
df.loc[df['revenue'].idxmax()]

budget                                                          237000000
genres                  [{"id": 28, "name": "Action"}, {"id": 12, "nam...
id                                                                  19995
keywords                [{"id": 1463, "name": "culture clash"}, {"id":...
original_language                                                      en
original_title                                                     Avatar
popularity                                                        150.438
production_companies    [{"name": "Ingenious Film Partners", "id": 289...
production_countries    [{"iso_3166_1": "US", "name": "United States o...
release_date                                                   2009-12-10
revenue                                                        2787965087
runtime                                                               162
spoken_languages        [{"iso_639_1": "en", "name": "English"}, {"iso...
status                                

In [116]:
maximum = df.loc[df['revenue'].idxmax()].original_title

In [117]:
maximum

'Avatar'

In [118]:
df.loc[df['revenue'].idxmin()]

budget                                                           27000000
genres                  [{"id": 28, "name": "Action"}, {"id": 12, "nam...
id                                                                  79698
keywords                                                               []
original_language                                                      en
original_title                                                 The Lovers
popularity                                                        2.41853
production_companies    [{"name": "Corsan", "id": 7299}, {"name": "Bli...
production_countries    [{"iso_3166_1": "AU", "name": "Australia"}, {"...
release_date                                                   2015-02-13
revenue                                                                 0
runtime                                                               109
spoken_languages                 [{"iso_639_1": "en", "name": "English"}]
status                                

In [119]:
minimun = df.loc[df['revenue'].idxmin()].original_title

In [120]:
minimun

'The Lovers'

ANS : Movie with the most earned revenue is 'Avatar' and
Movie with the least earned revenue is 'The Lovers'

# 5) What is the average runtime of movies in the year 2006?

In [121]:
start_date = "2006-01-01"
end_date = "2007-01-01"

In [122]:
start_d = df['release_date'] >= start_date

In [123]:
end_d = df['release_date'] <= end_date

In [124]:
dates_bet = start_d & end_d

In [125]:
filter_d = df.loc[dates_bet]

In [126]:
filter_d

budget                                             genres      id  \
10    270000000  [{"id": 12, "name": "Adventure"}, {"id": 14, "...    1452   
12    200000000  [{"id": 12, "name": "Adventure"}, {"id": 14, "...      58   
33    210000000  [{"id": 12, "name": "Adventure"}, {"id": 28, "...   36668   
104   160000000  [{"id": 12, "name": "Adventure"}, {"id": 28, "...     503   
139   150000000  [{"id": 12, "name": "Adventure"}, {"id": 28, "...     956   
...         ...                                                ...     ...   
4612     300000                      [{"id": 18, "name": "Drama"}]   26518   
4662          0                     [{"id": 35, "name": "Comedy"}]   40963   
4722     100000                      [{"id": 18, "name": "Drama"}]   18925   
4754          0  [{"id": 18, "name": "Drama"}, {"id": 35, "name...   84659   
4791         13                     [{"id": 27, "name": "Horror"}]  157185   

                                               keywords original_language  \
10    [{"id": 83, "name": "saving the world"}, {"id"...                en   
12    [{"id": 616, "name": "witch"}, {"id": 663, "na...                en   
33    [{"id": 1852, "name": "mutant"}, {"id": 8828, ...                en   
104   [{"id": 613, "name": "new year's eve"}, {"id":...                en   
139   [{"id": 220, "name": "berlin"}, {"id": 591, "n...                en   
...                                                 ...               ...   
4612         [{"id": 187056, "name": "woman director"}]                en   
4662        [{"id": 10183, "name": "independent film"}]                en   
4722  [{"id": 5950, "name": "christian"}, {"id": 607...                en   
4754                                                 []                en   
4791           [{"id": 14903, "name": "home invasion"}]                en   

                                  original_title  popularity  \
10                              Superman Returns   57.925623   
12    Pirates of the Caribbean: Dead Man's Chest  145.847379   
33                         X-Men: The Last Stand    3.857526   
104                                     Poseidon   21.133748   
139                      Mission: Impossible III   63.079003   
...                                          ...         ...   
4612                                     Old Joy    1.673208   
4662                              Little Big Top    0.092100   
4722                           Facing the Giants    4.563698   
4754                               The Naked Ape    0.077577   
4791                                 Tin Can Man    0.332679   

                                   production_companies  \
10    [{"name": "DC Comics", "id": 429}, {"name": "L...   
12    [{"name": "Walt Disney Pictures", "id": 2}, {"...   
33    [{"name": "Ingenious Film Partners", "id": 289...   
104   [{"name": "Virtual Studios", "id": 449}, {"nam...   
139   [{"name": "Paramount Pictures", "id": 4}, {"na...   
...                                                 ...   
4612  [{"name": "Film Science", "id": 1976}, {"name"...   
4662          [{"name": "Fly High Films", "id": 24248}]   
4722         [{"name": "Destination Films", "id": 769}]   
4754                                                 []   
4791  [{"name": "Park Films", "id": 21871}, {"name":...   

                                   production_countries release_date  \
10    [{"iso_3166_1": "US", "name": "United States o...   2006-06-28   
12    [{"iso_3166_1": "JM", "name": "Jamaica"}, {"is...   2006-06-20   
33    [{"iso_3166_1": "CA", "name": "Canada"}, {"iso...   2006-05-24   
104   [{"iso_3166_1": "US", "name": "United States o...   2006-05-12   
139   [{"iso_3166_1": "CN", "name": "China"}, {"iso_...   2006-05-03   
...                                                 ...          ...   
4612  [{"iso_3166_1": "US", "name": "United States o...   2006-01-01   
4662  [{"iso_3166_1": "US", "name": "United States o...   2006-01-01   
4722  [{"iso_3166_1":

In [127]:
new_d = filter_d

In [128]:
new_d.runtime.describe()

count    243.000000
mean     104.897119
std       22.223362
min        0.000000
25%       91.500000
50%      103.000000
75%      117.000000
max      193.000000
Name: runtime, dtype: float64

In [129]:
mean_d = new_d["runtime"].mean()

In [130]:
mean_d 

104.89711934156378

ANS : The average runtime of movies in the year 2006 is *104.89711934156378*